src/ui/jexlParser.js

```diff
@@ -0,0 +1,176 @@
+import Lexer from "mozjexl/lib/Lexer";
+import { elements as grammar } from "mozjexl/lib/grammar";
+import Parser from "mozjexl/lib/parser/Parser";
+
+/* eslint-disable */
+
+export async function evaluateJEXL(expression, context = {}) {
+  const lexer = new Lexer(grammar);
+  const parser = new Parser(grammar);
+
+  parser.addTokens(lexer.tokenize(expression));
+  const ast = parser.complete();
+  const falseParts = [];
+
+  async function evaluateExpression(expr) {
```

Comment:
Can you add documentation to all these fns explainign what they do?


They don't need to live inside the `evaluateJEXL` method -- they can be top level.